engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/BeanMetaDataImpl.java

```diff
@@ -175,7 +176,15 @@ public BeanMetaDataImpl(Class<T> beanClass,
 				cascadedProperties.add( propertyMetaData );
 			}
 			else {
-				allMetaConstraints.addAll( propertyMetaData.getTypeArgumentsConstraints() );
+				Set<MetaConstraint<?>> typeConstraints = propertyMetaData.getTypeArgumentsConstraints().stream()
```

Comment:
Ideally, we'd get rid of the special handling of type parameter constraints altogether. But my preference is to leave this as is for now and look into it again when implementing the container stuff proper from BV 2.0. The delegate business in `TypeParameterLocation` is kind of a hack, but I think it's good enough for now and allows us to move on.